src/Controls/src/Core/Shell/Shell.cs

```diff
@@ -1605,6 +1605,37 @@ void SendNavigated(ShellNavigatedEventArgs args)
 			CurrentItem?.Handler?.UpdateValue(Shell.TabBarIsVisibleProperty.PropertyName);
 		}
 
+		void PropagateSendNavigatedTo()
+		{
+			if (CurrentPage is null)
+			{
+				return;
+			}
+
+			// On Windows, the Loaded event has already fired (IsLoaded=true), so SendNavigatedTo runs immediately without altering the flow.
+			if (CurrentPage.IsLoaded)
+			{
+				CurrentPage.SendNavigatedTo(new NavigatedToEventArgs(_previousPage));
+			}
+			else
+			{
+				void OnCurrentPageLoaded(object sender, EventArgs e)
+				{
+					if (sender is Page page)
+					{
+						page.Loaded -= OnCurrentPageLoaded;
+						page.SendNavigatedTo(new NavigatedToEventArgs(_previousPage));
+
+						// Restore flyout behavior observers after deferred NavigatedTo timing
+						// Android requires this call to maintain flyout functionality
+						CurrentContent?.EvaluateDisconnect();
```

Comment:
The comment says "Android requires this call" but there's no platform check. This might have unintended side effects on iOS?